app/com/arpnetworking/metrics/portal/alerts/impl/DailyPartitionCreator.java

```diff
@@ -131,49 +135,59 @@ public static Props props(
+     * @throws InterruptedException if the actor does not stop within the allotted timeout, or if the actor thread was
      * interrupted for other reasons.
      */
-    public static void start(
-            final ActorRef ref,
-            final Duration timeout
-    ) throws ExecutionException, InterruptedException {
-        Patterns.ask(
-                ref,
-                START_TICKING,
-                timeout
-        )
-                .thenCompose(reply -> {
-                    @SuppressWarnings("unchecked") final Optional<Throwable> o = (Optional<Throwable>) reply;
-                    final CompletableFuture<Void> future = new CompletableFuture<>();
-                    o.ifPresent(future::completeExceptionally);
-                    future.complete(null);
-                    return future;
-                })
-                .toCompletableFuture()
-                .get();
+    public static void stop(final ActorRef ref, final Duration timeout) throws ExecutionException, InterruptedException {
+        Patterns.gracefulStop(ref, timeout).toCompletableFuture().get();
     }
 
     /**
-     * Ask the actor referenced by {@code ref} to stop execution.
+     * Ask the actor referenced by {@code ref} to create the partition(s) needed
+     * for the given date.
      *
      * @param ref an {@code DailyPartitionCreator}.
+     * @param date The partition date
      * @param timeout timeout for the operation
      * @throws ExecutionException if an exception was thrown during execution.
-     * @throws InterruptedException if the actor does not stop within the allotted timeout, or if the actor thread was
+     * @throws InterruptedException if the actor does not respond within the allotted timeout, or if the actor thread was
      * interrupted for other reasons.
      */
-    public static void stop(final ActorRef ref, final Duration timeout) throws ExecutionException, InterruptedException {
-        Patterns.gracefulStop(ref, timeout).toCompletableFuture().get();
+    public static void ensurePartitionExistsForDate(
+            final ActorRef ref,
+            final LocalDate date,
+            final Duration timeout
+    ) throws ExecutionException, InterruptedException {
+        final CreateForRange createPartitions = new CreateForRange.Builder()
+                .setStart(date)
+                .setEnd(date.plusDays(1))
+                .build();
+        Patterns.ask(
+                ref,
+                createPartitions,
+                timeout
+        )
+        .toCompletableFuture()
+        .get();
     }
 
+    @Override
+    public void preStart() {
+        LOGGER.info().setMessage("Starting execution timer")
+                .addData("schema", _schema)
+                .addData("table", _table)
+                .addData("lookahead", _lookahead)
+                .log();
+        getSelf().tell(TICK, getSelf());
+        getTimers().startPeriodicTimer(TICKER_NAME, TICK, TICK_INTERVAL);
```

Comment:
I assume this gets cancelled by the base class `AbstractActorWithTimers`?

Reply:
yes they're lifecycle bound